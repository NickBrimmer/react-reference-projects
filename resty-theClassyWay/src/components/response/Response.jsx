import React from 'react';
import PropTypes from 'prop-types';

const Response = ({ jsonResponse }) => (
  <pre>
    {JSON.stringify(jsonResponse, null, 2)}
  </pre>
);

Response.propTypes = {
  jsonResponse: PropTypes.oneOf([PropTypes.array, PropTypes.object])
};

export default Response;
