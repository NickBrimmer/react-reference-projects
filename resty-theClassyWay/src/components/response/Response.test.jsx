import React from 'react';
import { shallow } from 'enzyme';
import Response from './Response';

describe('Response component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Response jsonResponse={{ name: 'spot', age: 5, wegiht: '20 lbs' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
