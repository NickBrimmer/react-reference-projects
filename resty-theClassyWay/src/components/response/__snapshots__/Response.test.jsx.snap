// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response component renders App 1`] = `
<pre>
  {"name":"spot","age":5,"wegiht":"20 lbs"}
</pre>
`;
